tests/system_test.py

```diff
@@ -126,7 +128,7 @@ def retry_delay(deadline, delay, max_delay):
 TIMEOUT = float(os.environ.get("QPID_SYSTEM_TEST_TIMEOUT", 60))
 
 
-def retry(function, timeout=TIMEOUT, delay=.001, max_delay=1):
+def retry(function: Callable[[], bool], timeout: float = TIMEOUT, delay: float = .001, max_delay: float = 1):
```

Comment:
Just a note about mypy. It only checks functions that have type annotations in their signature. Anything that is not annotated is not checked. This is how adding annotations expands the fraction of the code that gets analyzed. It also means that gradually adding types somewhere may reveal type error somewhere else entirely. Therefore, when adding type annotations "on the side" like this, one needs to tread lightly, otherwise one ends up in typing hell, instead of just doing the originally intended task.